### PR TITLE
vendor: vendor `bazel-remote` project

### PR DESCRIPTION
**Pull request by rickystewart**
We want to integrate this with `dev`.

Release note: None

### CODE REVIEW

File: DEPS.bzl [hunk 1]
```diff
@@ -934,10 +934,10 @@ def go_deps():
         name = "com_github_census_instrumentation_opencensus_proto",
         build_file_proto_mode = "disable_global",
         importpath = "github.com/census-instrumentation/opencensus-proto",
-        sha256 = "579ab1ff19403579c3b438ab95528dfc691f0ff6076829f4ed2b2d2dc5b72552",
-        strip_prefix = "github.com/census-instrumentation/opencensus-proto@v0.3.0",
+        sha256 = "b3c09f3e635d47b4138695a547d1f2c7138f382cbe5a8b5865b66a8e08233461",
+        strip_prefix = "github.com/census-instrumentation/opencensus-proto@v0.2.1",
```
Comment (rickystewart) [inline DEPS.bzl:938]: Looks like `go get` downgraded this package. Not sure how serious that is.

Comment (dt) [reply DEPS.bzl:978]: I'm not quite sure why we're even still pulling it in in bazel, since go.mod/sum dropped it entirely didn't they?

Comment (dt) [reply DEPS.bzl:978]: I see: we dropped the dep entirely on the go side, but on the bazel side we're still picking it up from https://github.com/cockroachdb/vendored/compare/88451783657a8e7c399cc3ae217f4f763d0ac707...7f7a154256f9cbe12f21d80c7abf514e9a094a11#diff-2a297da5931dd3fecdd5c37c2ddaeb3ed2eb521d84db6bdb841f8e1bf5174f69R318 so then bazel is picking the lower version requested for the now transitive-only dep.

**Review by dt — APPROVED** [review]
I don't fully understand why go mod doesn't think we need that downgraded transitive dep anymore but if it's happy then who am I to argue.

**Comment by rickystewart** [comment]
TFTR!

bors r=dt

**Pull request merged**